Statistics/Histograms/Histogram.cs

```diff
@@ -446,57 +481,51 @@ public double InverseCDF(double p)
                 
             }
         }
-        public void AddHistograms(List<IHistogram> histograms)
+        public static IHistogram AddHistograms(List<IHistogram> histograms)
         {
-            //if histograms.count == 0, then nothing happens, as you would expect
-            //adding 0 histograms to *this* histogram changes nothing 
+            IHistogram aggregatedHistogram;
+
             if (histograms.Count > 0)
             {
                 ConvergenceCriteria convergenceCriteria = histograms[0].ConvergenceCriteria;
                 double min = 0;
                 double max = 0;
-                int sampleSize = 0;
-                foreach (Histogram histogramToAdd in histograms)
+                int binQuantity = 0;
+                foreach (IHistogram histogramToAdd in histograms)
                 {
-                    double newMin = Math.Min(min, histogramToAdd.Min);
-                    min = newMin;
-                    double newMax = Math.Max(max, histogramToAdd.Max);
-                    max = newMax;
-                    int newSampleSize = Math.Max(sampleSize, (int)histogramToAdd.SampleSize);
-                    sampleSize = newSampleSize;
+                    min += histogramToAdd.Min;
+                    max += histogramToAdd.Max;
+                    binQuantity = Math.Max(binQuantity, histogramToAdd.BinCounts.Length);
                 }
                 double range = max - min;
-                double binQuantity = 1 + 3.322 * Math.Log(sampleSize); //sturges rule 
-                double binWidth = range / sampleSize;
-                Histogram histogram = new Histogram(min, binWidth, convergenceCriteria);
-                int seed = 1234;
-                Random random = new Random(seed);
-                for (int i = 0; i < sampleSize; i++)
+                //double binQuantity = 1 + 3.322 * Math.Log(sampleSize); //sturges rule 
+                double binWidth = range / binQuantity;
+                aggregatedHistogram = new Histogram(min, binWidth, convergenceCriteria);
+
+                for (int i = 0; i < binQuantity; i++)
                 {
+                    double probabilityStep = (i + 0.5) / binQuantity;
```

Comment:
stepping across the probability domain on each of the histograms to get each of their InvCDFs at a given probability which are then added. I think doing i + 0.5 gets us a "nicer" cumulative probability - like the cumulative probability up to the middle of a bin. The really hard thing about this math problem is that BinCounts.Length is different for each histogram - otherwise we could stack the histograms like pancakes.